### PR TITLE
Adding functionality

### PR DESCRIPTION
**Pull request by ArinaSavchenko**

### CODE REVIEW

File: ClientApp/src/app/Flight.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import {Airport} from './Airport';
+import {Time} from '@angular/common';
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/Flight.ts]: What do you need these imports for?

File: ClientApp/src/app/Shared.service.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    date: new Date(),
+    amount: 1
+  };
+
+  empty2: FlightForSearch  = {
+    departure: 0,
+    arrival: 0,
+    date: new Date(),
+    amount: 1
+  };
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/Shared.service.ts:24]: What do you need these empty objects for?

File: ClientApp/src/app/flight-search-results/flight-search-results.component.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import {Observable, of, Subject} from 'rxjs';
+import {
+  debounceTime, distinctUntilChanged, switchMap
+} from 'rxjs/operators';
+
+import { Flight } from '../Flight';
+import { FlightService } from '../flight.service';
+import { SharedService } from '../Shared.service';
+import {FlightForSearch} from '../FlightForSearch';
+import {removeSummaryDuplicates} from '@angular/compiler';
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/flight-search-results/flight-search-results.component.ts]: This import should be along with other angular imports. Also do you even need it?

File: ClientApp/src/app/flight.service.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  getFlights(): Observable<Flight[]>{
+    return this.http.get<Flight[]>(this.flightsUrl);
+  }
+
+  searchFlights(flight: FlightForSearch): Observable<Flight[]>{
+    if (!flight.departure || !flight.arrival){
+      return of([]);
+    }
+    return  this.http.get<Flight[]>(`${this.flightsUrl}/?departure=${flight.departure}&&arrival=${flight.arrival}&&date=${flight.date}`);
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/flight.service.ts]: Use HttpParams https://angular.io/guide/http#configuring-http-url-parameters

File: ClientApp/src/app/flights-search-bar/flights-search-bar.component.css [hunk 5]
```diff
@@ -67,3 +61,19 @@
   background-color: #bcd30b;
   width: 30%;
 }
+
+.airports{
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/flights-search-bar/flights-search-bar.component.css]: Here and everywhere else in css: please use space before the opening bracket.

File: ClientApp/src/app/flights-search-bar/flights-search-bar.component.html [hunk 6]
```diff
@@ -1,56 +1,68 @@
...
-    [(ngModel)]="ticketsType">
-    <mat-radio-button value="1">One way</mat-radio-button>
-    <mat-radio-button value="2">Return</mat-radio-button>
-  </mat-radio-group>
+    <mat-radio-group
+      aria-label="Select an option"
+      class="radio_buttons"
+      (ngModelChange)="OnTicketTypeChange($event)"
+      formControlName="ticketType">
+      <mat-radio-button value="1" >One way</mat-radio-button>
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/flights-search-bar/flights-search-bar.component.html]: unnecessary space

File: ClientApp/src/app/flights-search-bar/flights-search-bar.component.html [hunk 7]
```diff
@@ -1,56 +1,68 @@
...
         <mat-label>Choose departure date</mat-label>
-        <input matInput [matDatepicker]="picker">
+        <input matInput [min]="today" [matDatepicker]="picker"
+               (ngModelChange)="setDateTo($event)" formControlName="dateTo">
         <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
         <mat-datepicker #picker></mat-datepicker>
       </mat-form-field>
 
-      <mat-form-field appearance="fill" class="datePicker" *ngIf="ticketsType == 2">
+      <mat-form-field appearance="fill" class="datePicker" *ngIf="+ticketType === 2">
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/flights-search-bar/flights-search-bar.component.html]: Avoid using magic numbers. Why is it even a number?

File: ClientApp/src/app/flights-search-bar/flights-search-bar.component.ts [hunk 8]
```diff
@@ -1,30 +1,96 @@
...
-import {FormControl} from '@angular/forms';
-import {Observable} from 'rxjs';
-import {map, startWith} from 'rxjs/operators';
+import {AfterContentInit, AfterViewChecked, ChangeDetectorRef, Component, OnInit} from '@angular/core';
+import {FormBuilder, FormControl, FormGroup, NgForm, PatternValidator, Validators} from '@angular/forms';
+
+import { FlightForSearch } from '../FlightForSearch';
+import { SharedService } from '../Shared.service';
+import { debounceTime, distinctUntilChanged, switchMap } from 'rxjs/operators';
+import { Observable, Subject } from 'rxjs';
```
Comment (andrei-kruhliakou-itechart) [inline ClientApp/src/app/flights-search-bar/flights-search-bar.component.ts:4]: These library imports should be with other third-party code

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]

File: ClientApp/src/app/Flight.ts (hunk 1 above)
Comment (ArinaSavchenko) [reply ClientApp/src/app/Flight.ts]: I've deleted Airport import as it's not needed anymore, but Time is required to set datatype Time to property flight_length

File: ClientApp/src/app/Shared.service.ts (hunk 2 above)
Comment (ArinaSavchenko) [reply ClientApp/src/app/Shared.service.ts:24]: They're needed to share info between components. They can't be null or undefined

**Review by andrei-kruhliakou-itechart — APPROVED** [review]

**Pull request merged**